x-pack/plugins/ingest_manager/server/services/epm/archive/index.ts

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import yaml from 'js-yaml';
+import { uniq } from 'lodash';
+import { title } from 'process';
+
+import {
+  ArchivePackage,
+  Dataset,
+  RegistryConfigTemplate,
+  RegistryInput,
+  RegistryStream,
+  RegistryVarsEntry,
+} from '../../../../common/types';
+import { PackageInvalidArchiveError, PackageUnsupportedMediaTypeError } from '../../../errors';
+import { cacheGet, cacheSet, setArchiveFilelist } from '../registry/cache';
+import { unzipBuffer, untarBuffer, ArchiveEntry } from '../registry/extract';
+
+export async function loadArchivePackage({
+  archiveBuffer,
+  contentType,
+}: {
+  archiveBuffer: Buffer;
+  contentType: string;
+}): Promise<{ paths: string[]; archivePackageInfo: ArchivePackage }> {
+  const paths = await unpackArchiveToCache(archiveBuffer, contentType);
+  const archivePackageInfo = parseAndVerifyArchive(paths);
+  setArchiveFilelist(archivePackageInfo.name, archivePackageInfo.version, paths);
+
+  return {
+    paths,
+    archivePackageInfo,
+  };
+}
+
+function getBufferExtractorForContentType(contentType: string) {
+  if (contentType === 'application/gzip') {
```

Comment:
We only support zip packages. Code for gzip should be removed.

Reply:
We still support tar.gz coming from a registry.

I'd rather remove suoport for tar.gz in both places at once, in a separate PR. It doesn't hurt to have both in at the moment, does it?